compat/mingw.c

```diff
@@ -3296,8 +3587,32 @@ static void setup_windows_environment(void)
 			struct strbuf buf = STRBUF_INIT;
 			strbuf_addstr(&buf, tmp);
 			if ((tmp = getenv("HOMEPATH"))) {
+				/*
+				 * Since Vista, HOMEDRIVE/HOMEPATH are set to Windows\System32
+				 * for secondary logons/services instead of being unset.
+				 * We don't want to (and likely can't) save stuff there
+				 * so ignore them in such a case.
+				 */
+				wchar_t w_system_dir[MAX_PATH];
+				wchar_t *w_buf;
 				strbuf_addstr(&buf, tmp);
-				if (is_directory(buf.buf))
+				{
+					UINT len_system_dir;
+					size_t len_w_buf;
+					if (!(len_system_dir = GetSystemDirectoryW(
```

Comment:
What's the purpose of `len_system_dir`? It gets assigned a value here and is then never used.

Reply:
Yes, it can be removed. It was used in a previous version of code.